1-Draft/RFCXXXX-PowerShell-in-Application-Mode.md

```diff
@@ -0,0 +1,118 @@
+﻿---
+RFC: RFCXXXX
+Author: Bruce Payette
+Status: Draft
+SupercededBy: 
+Version: 6.?
+Area: Hosting
+Comments Due: 01/03/2019
+Plan to implement: Yes
+---
+
+#PowerShell Application Mode
+
+##Motivation
+
+There is increasing need for the ability to deploy stand-alone non-interactive (embedded) PowerShell “applications”. These applications would be consumed by higher-layer services which, for various reasons, can’t run PowerShell in-process with the existing PowerShell API.
+Examples:
+
+-	Azure Functions/AWS lambdas
+-	Azure Stack
+-	PowerShell language service for VSCode, etc.
+-	Azure resource extension scripts/configurations
+-	Puppet/Chef
+
+##Specifcation
+
+This type of PowerShell deployment differs from traditional interactive applications in a number if ways:
+-	The functionality is likely predefined – the application is composed of a known set of scripts and modules.
+-	This implies that only a subset of the PowerShell functionality is required. For example, the help and formatting subsystems might be unnecessary. The net result would be faster startup, reduced footprint and consequently a reduced security Surface.
```

Comment:
As for Help System, last days I am trying decouple it to separate dll and module from engine. I believe this is more flexible.  Progress is about 80%. There is some problems with internal API problems that can be solved step-by-step PRs.  I am ready to push initial PR.